modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -153,22 +149,19 @@ private AQueryBuilder q() {
   }
 
   /**
-   * Create an authorization predicate to be used with {@link #isAuthorized(Predicate)},
+   * Create an authorization predicate to be used with {@link #isAuthorized(String, String)},
    * restricting access to the user's organization and the given action.
    *
    * @param action
    *     the action to restrict access to
    */
   private Predicate mkAuthPredicate(final String action) {
     final AQueryBuilder q = q();
-    return $(secSvc.getUser().getRoles())
-        .foldl(q.always().not(),
-               new Fn2<Predicate, Role, Predicate>() {
-                 @Override public Predicate apply(Predicate predicate, Role role) {
-                   return predicate.or(mkSecurityProperty(q, role.getName(), action).eq(true));
-                 }
-               })
-        .and(restrictToUsersOrganization());
+    return secSvc.getUser().getRoles().stream()
+            .map((role) -> mkSecurityProperty(q, role.getName(), action).eq(true))
+            .reduce(Predicate::or)
+            .orElseGet(() -> q.always().not())
```

Comment:
Only add a never matching predicate if there are no role checks because a user has no roles.

In theory, we could even stop here since we already know that the query will never return anything. We wouldn't even need to run that query. But that would require more changes and it is actually very unlikely to happen since every user should usually have at least his own user role.